internal/services/applications/parse/optional_claims.go

```diff
@@ -66,3 +66,13 @@ func (id OptionalClaimsId) Segments() []resourceids.Segment {
 func (id OptionalClaimsId) String() string {
 	return fmt.Sprintf("Application Optional Claims (Application ID: %q)", id.ApplicationId)
 }
+
+func (id OptionalClaimsId) FromParseResult(input resourceids.ParseResult) error {
```

Comment:
This'll need to be a pointer:

```suggestion
func (id *OptionalClaimsId) FromParseResult(input resourceids.ParseResult) error {
```